### ISSUE TITLE
Week comparison feature

### ISSUE THREAD

**Issue by tautautau** [wbish/jsondiffpatch.net#18]
Hi.

Have you thought about week comparison of the json?
For example :
expected: {"key1":"value1"}

given: {"key1":"value1", "key2":"value2"}
Would pass.

given: {"key1":"valueX", "key2":"value2"}
Would fail.

It basically wouldn't take in to account elements which are not in the expected string. If there is a need I could help you with that feature.

**Comment by wbish** [comment]
> expected: {"key1":"value1"}
> 
> given: {"key1":"value1", "key2":"value2"}
> Would pass.

I'm not sure I fully understand how it would tie together.

When you say pass, do you mean the Diff method would return null as if they are equal?

You illustrate what happens when left does not have key2, but what happens in the case the right object does not have key1? Does that get treated as a delete of the key or just ignored?

**Comment by tautautau** [comment]
Sorry for late replay.

> When you say pass, do you mean the Diff method would return null as if they are equal?

Yes.

> You illustrate what happens when left does not have key2, but what happens in the case the right object does not have key1? Does that get treated as a delete of the key or just ignored?

If it would not have key1 it would be treated as deleted. 

Basically I'm using jsondiffpatch for integration tests. Sometimes there is such case when expected json has dynamic values which depends on current time, or dynamic links which depends on what server you run the test. I would like to have those not compared. Or just to check if the key exists but value ignored.

I did some coding in java world and I used https://github.com/skyscreamer/JSONassert. I was kind of fan of it. There are some examples: http://jsonassert.skyscreamer.org/cookbook.html. 

I think the possibility to have weak/strict comparison would be really useful :)
